[CI] Alternative build/test without lix/vshaxe-build

kLabz opened the pull request:
* No longer removing lix/vshaxe-build, just providing alternative
* Outdated libs should make CI fail pretty much all the time without adding another check
* Updating `install.hxml` locally can be done with [haxe-manager](https://github.com/kLabz/haxe-manager)'s `hx lix-to-install install.hxml`
* CI is now also running on latest Haxe nightlies to detect issues earlier
* CI is now also running on mac ~~and windows~~ (CI is [disabled](https://github.com/vshaxe/haxe-language-server/pull/122/commits/6c261595701b2455909dd9d7062451973ef19eab) on windows for now because tests have been broken there for a while)
* Note: had to [increase timeout](https://github.com/vshaxe/haxe-language-server/pull/122/commits/5e598b89c309936645e8b9910cc1ad4d281b3735) on a few tests because of slow macos runners hitting timeout there

AlexHaxe commented:
I don't mind getting rid of vshaxe-build.

dropping lix is a step backwards for me. not only does it break my workflow, we also lose the ability to lock local compilation to a specific Haxe version.

kLabz commented:
I don't mind keeping lix and non-lix in sync myself (and adding a CI check) but I do want a non-lix workflow to be available too

AlexHaxe commented:
I'm fine with that.

I'll try to remember updating both when adding or upgrading dependencies. if that can be ensured via a CI check that would definitely help keeping them in sync.

kLabz commented:
Updated the PR:

* Updated base branch and libs versions
* No longer removing lix/vshaxe-build, just providing alternative
* Outdated libs should make CI fail pretty much all the time without adding another check
* Updating `install.hxml` locally can be done with [haxe-manager](https://github.com/kLabz/haxe-manager)'s `hx lix-to-install install.hxml`

Current issues are "unrelated":

* mac job fails because of https://github.com/krdlab/setup-haxe/issues/53
* windows tests fail but that's not new; was just not running before
